pip/basecommand.py

```diff
@@ -314,6 +314,23 @@ def populate_requirement_set(requirement_set, args, options, finder,
                     'You must give at least one requirement to %(name)s '
                     '(see "pip help %(name)s")' % opts)
 
+        # On Windows, any operation modifying pip should be run as:
+        #     python -m pip ...
+        # See https://github.com/pypa/pip/issues/1299 for more discussion
+        should_show_use_python_msg = (
+            sys.platform == 'win32' and
```

Comment:
This checks fails at least on Jython and IronPython. I've used `os.sep != '/'` which is hackish but works.